rsyslog/openstack-logs-to-kafka-over-director.conf.example

```diff
@@ -0,0 +1,59 @@
+module(load="imfile" PollingInterval="10")
+
+$MaxMessageSize 92k
+
+# pacemaker_system.pacemaker
+input(type="imfile"
+  file="/var/log/host/pacemaker/pacemaker.log"
+  startmsg.regex="^[a-zA-Z]{3} [0-9]{2} [:0-9]{8}"
+  tag="system.pacemaker"
+  ruleset="kafka-logging"
+  persiststateinterval="1"
+  deleteStateOnFileDelete="off"
+  freshStartTail="on"
+  reopenOnTruncate="on"
+)
+
+input(type="imfile"
+  file="/var/log/containers/*/*.log"
+  tag="openstack"
+  startmsg.regex="^[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}(.[0-9]+ [0-9]+)? (INFO|WARNING|ERROR) "
+  ruleset="kafka-logging"
+  persiststateinterval="1"
+  deleteStateOnFileDelete="off"
+  freshStartTail="on"
+  reopenOnTruncate="on"
+)
+
+
+template (name="rsyslog-record" type="list"
+ option.jsonf="on")
+{
+    property(dateFormat="rfc3339" format="jsonf" name="timereported" outname="@timestamp" )
+    property(format="jsonf" name="hostname" outname="host" )
+    property(format="jsonf" name="syslogseverity" outname="severity" )
+    property(format="jsonf" name="syslogfacility-text" outname="facility" )
+    property(format="jsonf" name="syslogtag" outname="tag" )
+    property(format="jsonf" name="app-name" outname="source" )
+    property(format="jsonf" name="msg" outname="message" )
+    property(format="jsonf" name="$!metadata!filename" outname="file" )
+    constant(format="jsonf" outname="cloud" value="<cloud_url>" )
+    constant(format="jsonf" outname="region" value="regionOne" )
+}
+
+ruleset(name="kafka-logging") {
+      action(type="omfwd" 
+             template="rsyslog-record"
+             queue.type="linkedlist"
+             queue.size="4000"
+             queue.dequeuebatchsize="200"
+#             queue.saveOnShutdown="on"
+             action.resumeRetryCount="-1"
+             keepAlive="on"
+             target="<director_fqdn>" 
+             port="514" 
+             protocol="tcp")
+}
+
+#$DebugFile /var/log/rsyslog.debug
```

Comment:
remove these lines if they are not required

Reply:
unnecessary comments removed by 77877616442f605b48b33480276c9e8098682ada